packages/inngest/src/components/connect/index.ts

```diff
@@ -254,126 +278,158 @@ class WebSocketWorkerConnection implements WorkerConnection {
       connect: "v1",
     };
 
-    const appName = this.inngest.id;
-
-    const functions: Array<FunctionConfig> = this.functions.flatMap((f) =>
-      f["getConfig"]({
-        baseUrl: new URL("wss://connect"),
-        appPrefix: appName,
-        isConnect: true,
-      })
-    );
+    const functionConfigs: Record<
+      string,
+      {
+        client: Inngest.Like;
+        functions: FunctionConfig[];
+      }
+    > = {};
+    for (const [appId, { client, functions }] of Object.entries(
+      this.functions
+    )) {
+      functionConfigs[appId] = {
+        client: client,
+        functions: functions.flatMap((f) =>
+          f["getConfig"]({
+            baseUrl: new URL("wss://connect"),
+            appPrefix: client.id,
+            isConnect: true,
+          })
```

Comment:
If you're feeling generous, a `getConnectConfig()` method that returns the correct data instead of adapting `getConfig()` would be awesome - `getConfig()` is soo old 😄